src/main/java/org/spongepowered/api/entity/living/player/Player.java

```diff
@@ -255,7 +255,7 @@ default JoinData getJoinData() {
      * @return True if played before, false otherwise
      */
     default boolean hasPlayedBefore() {
-        return !firstPlayed().equals(lastPlayed());
+        return !firstPlayed().get().equals(lastPlayed().get());
```

Comment:
You add the .gets, do Values not use deep equals?

Reply:
They do, and I have no idea why this was not working on my dev environment.

Reply:
It's more than likely because the default values are differing. Troubleshoot through with AbstractBaseValue.

Reply:
That would make sense.